src/library/basesqltablemodel.cpp

```diff
@@ -419,6 +441,30 @@ void BaseSqlTableModel::setSort(int column, Qt::SortOrder order) {
             m_sortColumns.removeLast();
         }
     }
+    
+    // Write new sortColumns order to user settings
+    QString val;
+    QTextStream out(&val);
+    for (SortColumn& sc : m_sortColumns) {
+
+        QString name;
+        if (sc.m_column > 0 && sc.m_column < m_tableColumns.size()) {
+            name = m_tableColumns[column];
+        } else {
+            int ccColumn = sc.m_column - m_tableColumns.size() + 1;
```

Comment:
Here ccColumn is 1 ... X 

But seams columnNameForFieldIndex to be zero based. 
```
inline QString columnNameForFieldIndex(int index) const {
        if (index < 0 || index >= m_columnsByIndex.size()) {
            return QString();
        }
        return m_columnsByIndex.at(index);
    }
```

Is it an issue?